client/src/pages/tasks/Show.js

```diff
@@ -90,11 +92,26 @@ export default class TaskShow extends Page {
 						<Form.Field id="shortName" label={`${taskShortLabel} number`} />
 						<Form.Field id="longName" label={`${taskShortLabel} description`} />
 						<Form.Field id="status" />
-						{task.responsibleOrg && task.responsibleOrg.id && this.renderOrg()}
-						<Form.Field id="customFieldEnum" label={`${customEnumLabel}`} />
-						<Form.Field label="Planned completion" id="plannedCompletion" value={task.plannedCompletion && moment(task.plannedCompletion).format('D MMM YYYY')} />
-						<Form.Field label="Projected completion" id="projectedCompletion" value={task.projectedCompletion && moment(task.projectedCompletion).format('D MMM YYYY')} />
-						<Form.Field id="customField" />
+
+						{task.responsibleOrg && task.responsibleOrg.id && 
+							this.renderOrg()
+						}
+						
+						{taskCustomEnumLabel &&
+						<Form.Field id="customFieldEnum" label={taskCustomEnumLabel} />
+						}
+
+						{taskPlannedCompletion &&
+							<Form.Field id="plannedCompletion" label={taskPlannedCompletion} value={task.plannedCompletion && moment(task.plannedCompletion).format('D MMM YYYY')} />
```

Comment:
Is it an idea to add the date formatting "D MMM YYY" in the dictionary?

Reply:
it is. Let's refactor the dictionary first and see how it goes